### ISSUE TITLE
Printing and plotting FFTs: Use default `data = "train"` and when to report `bacc` vs. `wacc`?

### ISSUE THREAD

**Issue by hneth** [ndphillips/FFTrees#72]
I like that `print.FFTrees()` now supports both training and test data, of course, but just noticed **2 more potential issues**:

1. As the default data setting is set to `data = "test"`, but running an R function automatically prints its output, running a harmless standard command, like

```{r}
FFTrees(survived ~., data = titanic)
```

presently yields the quite alarming **warning**:

```{r}
Warning message:
In print.FFTrees(x) :
  You asked to print 'test' data, but there were no test data. I'll print the best training tree instead...
```

Thus, in the absence of test data, the user _always_ gets warned when printing an object without explicitly setting `data = "train"` — even though s/he didn't even ask for printing the object! 

2. The present defaults for both `goal` and `goal.chase` are set to `wacc`. Thus, it seems odd that `print.FFTrees()` fails to report this measure.  (Whenever the default weighting of `sens.w = .50` is used, `wacc` is identical to `bacc`, of course.  But are users always aware of that?)

Both issues are rather minor and could easily be addressed.

**Suggestions**

1. As training data is always available, set the default in printing to `data = "train"` (and consider doing so in `plot.FFTrees()` as well...)

2. Include `wacc` by either 
    1. reporting both `bacc` _and_ `wacc` (by adding another line to the current report of accuracy measures and risking that the latter is mostly redundant),  or 
    2. replacing `bacc` by `wacc` in the current report.

But as I may be overlooking something here, I'd love to hear any ideas or suggestions before changing the code!

**Comment by ndphillips** [comment]
1. 100% agree. Let's change the default on both `print.FFTrees()` and `plot.FFTrees()` to `data = "train"`. My original thinking was that I was so worried about people focusing too much on training performance instead of testing. But now I see the user-experience issues that causes.

2. Good points. I think we should report `wacc` in `print.FFTrees()`.

**Comment by hneth** [comment]
**Thanks** for your swift reply.  I just implemented both of these changes prior to closing PR #71 (and changed the documentation of `print.FFTrees` and `plotFFTrees()` accordingly).

btw:  I also just noticed that `plotFFTrees()` is also showing `bacc`, rather than `wacc` in the bottom panel.  Perhaps we may want to switch to using `wacc` there as well?  Alternatively, we could also keep showing `bacc` as long as `sens.w = 0.50` and only switch to `wacc` when `sens.w` is set to a different value?

As most users may never deviate from the default setting (of `sens.w = .50`), the distinction between `bacc` and `wacc` may rarely matter.  Nevertheless, it seems sensible to handle it consistently across printing and plotting.

**Comment by hneth** [comment]
After thinking a bit more about the `bacc` vs. `wacc` issue, I propose the following:

**Suggestion**

Since `bacc` is a special, but probably by far the most common case of `wacc`, I suggest consistently using and reporting `bacc` (in `goal` and whenever printing and plotting FFTs) _unless_ `sens.w` has been set to a value that differs from the `bacc` default of 0.50.  *Iff* this is the case, `wacc` and `sens.w` should be used and reported throughout.

(The alternative of consistently using `wacc` seems unwarranted when `sens.w = 0.50` and would always require reporting the value of `sens.w` as well.) 

What do you think?

**Comment by hneth** [comment]
Upon further reflection, here is a more precise description of the issue and a suggested solution:

**Issue**: `bacc` is a special case of `wacc`. Specifically, when `sens.w = 0.50` (as per default), `wacc` _is_ `bacc`. 

**Suggestion**: Implement 2 types of changes:

1. _Use_: When `sens.w = 0.50` (as per default), correct any use of `wacc` when setting `goal`, `goal.chase`, or `goal.threshold` to `bacc` (and issue a corresponding feedback message).  When `sens.w` _differs_ from the default value, however, setting those parameters to `bacc` or `wacc` must _not_ be restricted (but `wacc` is used by default). 

2. _Display_: In all computations and displays of accuracy (e.g. when printing or plotting trees):  
When `sens.w = 0.50` (as per default), display only `bacc`, else display only `wacc` (and current `sens.w` value).

**Comment by hneth** [comment]
Issue has been resolved in PR #79.
